### PR TITLE
Client Side Validation and System Fields on CSV Export

### PR DESCRIPTION
**Pull request by ufoloko**
I have added client side validation for almost all types. #55 

Types with validation are are: 

Calculated Column

- Currency
- Date
- Date and Time
- Decimal
- Email (format needs regular expression)
- Integer
- Text
- Time
- RichHtml (plain mode), rich text was not possible.

Is tested for all those field types.

Additionally I have added an option to export system fields for CSV Export. this adds Created by, Created at, Changed by and Changed at fields. #16

### CODE REVIEW

File: DataTypes/Decimal.cs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using DotNetNuke.Services.Localization;
```
Comment (valadas) [inline DataTypes/Decimal.cs:1]:
```suggestion
```

File: DataTypes/Decimal.cs [hunk 2]
```diff
@@ -53,6 +54,7 @@ void EditDecimal_Init(object sender, EventArgs e)
         {
             CtlValueBox.Attributes.Add("style", "text-align:right");
             CtlValueBox.Width = new Unit("10em");
+
```
Comment (valadas) [inline DataTypes/Decimal.cs:57]:
```suggestion
```

File: DataTypes/Int32.cs [hunk 3]
```diff
@@ -1,3 +1,4 @@
+using DotNetNuke.Services.Localization;
```
Comment (valadas) [inline DataTypes/Int32.cs:1]:
```suggestion
```

File: DataTypes/Int32.cs [hunk 4]
```diff
@@ -56,6 +57,7 @@ protected override void OnInit(EventArgs e)
             base.OnInit(e); 
             CtlValueBox.Attributes.Add("style", "text-align:right");
             CtlValueBox.Width = new Unit("10em");
+
```
Comment (valadas) [inline DataTypes/Int32.cs:60]:
```suggestion
```

File: FormAndList.csproj [hunk 5]
```diff
@@ -371,6 +371,13 @@
     <Content Include="Providers\DataProviders\sqldataprovider\06.00.04.sqldataprovider" />
     <None Include="Providers\DataProviders\sqldataprovider\uninstall.sqldataprovider" />
     <Content Include="ShowXml.ashx" />
+    <Content Include="web.config" />
+    <None Include="web.Debug.config">
+      <DependentUpon>web.config</DependentUpon>
+    </None>
+    <None Include="web.Release.config">
+      <DependentUpon>web.config</DependentUpon>
+    </None>
```
Comment (valadas) [inline FormAndList.csproj:380]: There is no web.config in this module

**Review by valadas — CHANGES_REQUESTED** [review]
Just 2-3 files that have changes but did not need them and a reference to a web.config that does not exist in the module. In a perfect works it would have been nice to have 1 PR per feature, but since they are a bit small features, that looks ok to me otherwise.

Thanks.

**Comment by valadas** [comment]
Oh, also, there is a conflict with Time.cs if you can rebase on the current development branch. Thanks.

**Comment by ufoloko** [comment]
Sorry, im a little bit new in this repo. Im agree with changes you propose, what is the next step ?

**Comment by valadas** [comment]
Ok, let me see if I can resolve that in the browser...

**Comment by valadas** [comment]
Arrg, nope the whole file shows up as a conflict. So here are the options :)

1. rebase or merge back from the current develop branch to be able to fix the conflict and the requested changes and push back which will fix this PR
2. Start over clean from the current develop branch and maybe take that opportunity to break it into 2 independed PRs for the two issues it solves.
3. I can fix this PR but I got incredibly late on other jobs due to Summit and other stuff so it wont happen very quick :)

**Pull request closed without merging**